mlir/include/mlir/Dialect/ArmSME/IR/ArmSMEOps.td

```diff
@@ -369,7 +369,15 @@ def TileStoreOp : ArmSME_Op<"tile_store"> {
 }
 
 def LoadTileSliceOp : ArmSME_Op<"load_tile_slice", [
-    AllTypesMatch<["tile", "result"]>
+    AllTypesMatch<["tile", "result"]>,
+    TypesMatchWith<
+      "mask has i1 element type and same shape as result",
+      "result", "mask",
+      "VectorType("
+        "VectorType::Builder("
+          "::llvm::cast<mlir::VectorType>($_self)"
+        ").dropDim(0).setElementType(IntegerType::get($_self.getContext(), 1))"
+      ")">,
```

Comment:
Note that this will work with the `HasMatchingMaskTypeConstraint` from #70657 (`OptionalTypesMatchWith` also as expected with non-optional types, so you don't need another constraint).

Reply:
Oh, I misread this. The summary does not match what this is doing "mask has i1 element type and same shape as result", it's not got the same shape, it's a slice of the result.

Reply:
Ah, this is similar to the tile_load constraint, forgot to update when copying. Fixed.